### ISSUE TITLE
Server side only certs - openssl client - wrong certificate type

### ISSUE THREAD

**Issue by ryanhagen** [pion/dtls#442]
### Your environment.
  - Version: 2.1.0
  - Browser: What?
  - tldr: Server side only certificates - using openssl dtls client - fails

### What did you do?
I've created production letsencrypt certs that are valid.

**I configure my listener like so:**

    certificate, err := util.LoadKeyAndCertificate(dtls_key, dtls_cert)
	util.Check(err)

	rootCertificate, err := util.LoadCertificate("getssl-docker/certs/isrgrootx1.pem")
	util.Check(err)
	certPool := x509.NewCertPool()
	cert, err := x509.ParseCertificate(rootCertificate.Certificate[0])
	util.Check(err)
	certPool.AddCert(cert)

	// Prepare the configuration of the DTLS connection
	config := &dtls.Config{
		Certificates: []tls.Certificate{*certificate, *rootCertificate},
		// Create timeout context for accepted connection.
		ConnectContextMaker: func() (context.Context, func()) {
			return context.WithTimeout(ctx, 30*time.Second)
		},
	}

**Test my server (openssl 3.0.2)**
    `openssl s_client -dtls1_2 -connect valid.dns.name.for.cert:port -CAfile lecerts/chain.pem -debug`

**Output:**
**Note the cert Verification: OK, yet the client expects a different certificate type.**

> CONNECTED(00000007)
> depth=2 C = US, O = Internet Security Research Group, CN = ISRG Root X1
> verify return:1
> depth=1 C = US, O = Let's Encrypt, CN = R3
> verify return:1
> depth=0 CN = valid.dns.name.for.cert
> verify return:1
> **0016900F01000000:error:0A00017F:SSL routines:tls_post_process_server_certificate:wrong certificate type:[ssl/statem/statem_clnt.c:1914](https://github.com/openssl/openssl/blob/openssl-3.0.2/ssl/statem/statem_clnt.c#L1914):**
> 
> Certificate chain
>  0 s:CN = valid.dns.name.for.cert
>    i:C = US, O = Let's Encrypt, CN = R3
>    a:PKEY: rsaEncryption, 4096 (bit); sigalg: RSA-SHA256
>    v:NotBefore: Mar 31 04:25:40 2022 GMT; NotAfter: Jun 29 04:25:39 2022 GMT
> 
> no peer certificate available
> 
> No client certificate CA names sent
>
> SSL handshake has read 2415 bytes and written 809 bytes
> **Verification: OK**
> 
> New, (NONE), Cipher is (NONE)
> Secure Renegotiation IS supported
> Compression: NONE
> Expansion: NONE
> No ALPN negotiated
> SSL-Session:
>     Protocol  : DTLSv1.2
>     Cipher    : 0000
>     Session-ID: 
>     Session-ID-ctx: 
>     Master-Key: 
>     PSK identity: None
>     PSK identity hint: None
>     SRP username: None
>     Start Time: 1649105831
>     Timeout   : 7200 (sec)
>     Verify return code: 0 (ok)
>     Extended master secret: no
> 

### What did you expect?
I expect a successful handshake

### What happened?
The last packet sent in this handshake is from the openssl client; Alert - Fatal - Illegal Parameter (47)

The openssl client rejects the certificate as "wrong type". Here's the exact line of openssl code that's faulting [https://github.com/openssl/openssl/blob/openssl-3.0.2/ssl/statem/statem_clnt.c#L1914](url)

**Comment by boaks** [comment]
If you append a [IP-Capture](https://github.com/eclipse/californium/wiki/Logs-and-IP-Capturing-%E2%80%90-How-To-Provide-The-Right-Information#ip-capturing), there may be a chance, to see what fails.

**Comment by daenney** [comment]
You're loading your TLS key and the root CA separately, but that doesn't mean that Go will build the certificate chain for you. Now you just have 2 disjoint certs in your pool. The first one being offered is now a root CA, which your client really can't do anything with and probably errors out.

You don't want to do it this way server side. You need to create a cert bundle with your server cert + any intermediate CA (without the root CA, the client should have that one already because otherwise it can't verify anything). Concatenate those into a file, and load that instead.
